core/embed/trezorhal/stm32u5/secret.c

```diff
@@ -69,8 +69,29 @@ secbool secret_bhk_locked(void) {
          sectrue;
 }
 
+static secbool secret_present(uint32_t offset, uint32_t len) {
+  uint8_t *optiga_secret =
+      (uint8_t *)flash_area_get_address(&SECRET_AREA, offset, len);
+
+  int optiga_secret_empty_bytes = 0;
+
+  for (int i = 0; i < len; i++) {
+    if (optiga_secret[i] == 0xFF) {
+      optiga_secret_empty_bytes++;
+    }
+  }
+  return sectrue * (optiga_secret_empty_bytes != len);
+}
+
 void secret_bhk_provision(void) {
   uint32_t secret[SECRET_BHK_LEN / sizeof(uint32_t)] = {0};
+
+  ensure(secret_verify_header(), "secret sector not properly initialized");
```

Comment:
If the secret sector is not properly initialized, this could potentially lead to device bricking, especially if someone else does not reinitialize the secret sector (currently handled by the board-loader). It might be a bit safer to reinitialize the secret sector here if it's invalid

Reply:
done here: https://github.com/trezor/trezor-firmware/commit/0153fba65a000ac74667dae0b7e52551bfc66d6e

note that i was counting on boardloader doing the reinitialization, as that is the place where bootloader downgrade protection needs to be written after the secret is deleted. Which is why i am still halting the execution after this gets re-initialized, to enforce going through boardloader and write the protection.

also, this should not really happen, if the secret gets deleted at any point, the device permanently loses optiga pairing data and thus cant authenticate as genuine anymore.

Reply:
i also moved the bootloader downgrade protection here: https://github.com/trezor/trezor-firmware/commit/a86b30b12bd514d967c623ece8248e6f5313b8d8

with the logic that if bootloader re-initializes the secret, the downgrade protection needs to be filled before the potential sd-card bootloader upgrades happens, otherwise it would be filled with new data already potentially allowing bootloader downgrade

Reply:
(https://github.com/trezor/trezor-firmware/pull/3553/commits/2b48ab400bf417774a94b03779ea94c65fe982a9 reverts wrongly commited makefile modification)